GoogleCloudPlatform/spring-cloud-gcp
# fix: add runtime hints for trace

~~**Builds on top of CI setup in #1933. Will rebase once its merged in.**~~

This pr:
- adds `TraceRuntimeHints` in trace autoconfig module and unit test for it. This allows native builds.
- adds `SampleRuntimeHints` this enables [pubsub trace](https://googlecloudplatform.github.io/spring-cloud-gcp//reference/html/index.html#pubsub-trace-instrumentation-experimental) portion of the sample to run after Native Build. not exactly sure why this is needed.
- adds `TestRuntimeHints` to allow logging portion of the test setup run as expect. (same as #1933)

CI test should run sample integration test.

To test run manually:
- run application, build with `mvn clean package -Pnative-sample-config -Pnative`, then run application with `./target/spring-cloud-gcp-trace-sample`. Then follow sample README to test functionality.
- run test: `mvn -PnativeTest clean test -Pnative-sample-config`
expect to get:
```
com.example.TraceSampleApplicationIntegrationTests > testTracesAreLoggedCorrectly() SUCCESSFUL


Test run finished after 18500 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[         1 tests found           ]
[         0 tests skipped         ]
[         1 tests started         ]
[         0 tests aborted         ]
[         1 tests successful      ]
[         0 tests failed          ]

```


UPDATE: start failing on ci and local after rebase. Need investigate.
